### ISSUE TITLE
404 on tenant routes when running HTTP tests

### ISSUE THREAD

**Issue by AtumG** [archtechx/tenancy#170]
I'm really enjoying this package so far and really appreciate all your hard work! 
I have run into something I can't figure out.

I would like to be able to write a feature test that sets up a Tenant in the database, creates a user and then attempts to login via an HTTP request similar to the [Laravel HTTP Tests Documentation](https://laravel.com/docs/5.7/http-tests#introduction)

**The Problem**
When the test sends the POST request to attempt a login - I receive a 404 on a route that is defined in the `routes/tenant.php` file.

The `Tenant::createFrom()` method is successfully creating the tenant in the central database, creating the tenants DB and creating a user with the correct credentials in the respective `Tenant`'s database. I can verify this by visiting the tenant site and logging in with my browser.

Is it possible that the `routes/tenant.php` file is not loaded automatically for testing?

~~~php

<?php

namespace Tests\Feature;
use App\Models\Tenant;
use Tests\TestCase;

class TenantRegistrationTest extends TestCase
{

    /** @test */
    public function a_new_tenant_can_login()
    {

        $fqdn = str_random(5).'-login.multi-tenant-app.test';

        $tenant = Tenant::createFrom($fqdn, 'name', 'bat@man.com', 'password');

       tenancy()->initializeTenancy($tenant->tenant);

        $this->withoutExceptionHandling();

        
        $form = $this->post('https://'.$fqdn.'/login', [
                                'email' => 'bat@man.com',
                                'password' => 'password'
                            ]);
        
        /** 
        *   Fails here - Response status code [404] is not a successful status code. Failed asserting 
        *   that false is true.
        **/
        $form->assertSuccessful();

    }
}
~~~

I have the following in `routes/tenant.php`
~~~php
    Route::get('login', 'Auth\LoginController@showLoginForm');
    Route::post('login', 'Auth\LoginController@login')->name('login');
    Route::post('logout', 'Auth\LoginController@logout')->name('logout');
~~~

`App\Models\Tenant.php`
~~~php
public static function createFrom($fqdn, $name, $email, $password = null): self
    {
        $tenant = StanclTenant::create([$fqdn]);

        tenancy()->init($fqdn);

        static::addRolesAndPermissions($fqdn);
        static::addDefaultSettings($fqdn);


        $admin = static::makeAdmin($fqdn, $name, $email, $password);

        tenancy()->end();

        return new self($tenant, $admin);
    }
~~~

**Comment by stancl** [comment]
> Is it possible that the `routes/tenant.php` file is not loaded automatically for testing?

The routes are loaded only on domains listed *not* listed in `tenancy.exempt_domains`. Is `localhost` or `127.0.0.1` by any chance in your exempt domains? And do you see your tenant routes in `artisan route:list`?

I'm thinking that I could get rid of the conditional loading, since the `PreventAccessFromTenantDomains` middleware makes it impossible to visit tenant routes on central domains and vice versa.

Try removing the `if` check in `vendor/stancl/tenancy/src/TenantRouteServiceProvider.php`: https://github.com/stancl/tenancy/blob/2.x/src/TenantRouteServiceProvider.php#L14.

**Comment by drbyte** [comment]
For me, removing the wrapping `if` lines, but keeping the code inside the condition, does appear to resolve it.

**Comment by drbyte** [comment]
Update: removing the wrapping `if` lines, but keeping the code inside the condition, does appear to resolve it.

**Comment by stancl** [comment]
I'll fix this as part of #169

**Comment by AtumG** [comment]
Removing the conditional loading solved the problem. I also now have all my tenant routes showing in `php artisan:list` 

Thank you both for the help!
